### PR TITLE
Sync NCP3.2.1 manifests, configmap and role

### PR DESCRIPTION
**Pull request by timdengyun**
Sync manifests, configmap and role yamls from NCP 3.2.1 branch

### CODE REVIEW

File: deploy/kubernetes/role.yaml [hunk 1]
```diff
@@ -44,3 +44,6 @@ rules:
 - apiGroups: [policy]
   resources: [podsecuritypolicies]
   verbs: [create, get, list, patch, update, watch, delete, use]
+- apiGroups: ['']
```
Comment (timdengyun) [inline deploy/kubernetes/role.yaml:47]: Add new resource "events" RBAC based on changes in manifest yaml changes of adding events for ncp-patch-role.

- apiGroups:
   - ""
   resources:
     - events
   verbs:
     - create
     - get
     - list
     - patch
     - update
     - delete

Comment (jwsui) [reply deploy/kubernetes/role.yaml:47]: Do we need to update csv/bundle image as manifests updated?

Comment (timdengyun) [reply deploy/kubernetes/role.yaml:47]: I think we need to do that also. we can put it in the another patch once this patched merged. I recalled that you told me we can generate bundles images from manifests automatically.

File: manifest/kubernetes/rhel/ncp-rhel.yaml [hunk 2]
```diff
@@ -704,12 +716,24 @@ spec:
...
+
+
+      # nodeSelector is configured by operator ncp-install CRD
+      nodeSelector:
+        {{.NcpNodeSelector}}
+
       # tolerations is configured by operator ncp-install CRD
       tolerations:
-        {{ .NcpTolerations }}
+        {{.NcpTolerations}}
```
Comment (heypnus) [inline manifest/kubernetes/rhel/ncp-rhel.yaml:735]: Did you test it? I'm not sure whether `{{.NcpTolerations}}` is equal to `{{ .NcpTolerations }}`

File: manifest/openshift4/coreos/ncp-openshift4.yaml [hunk 3]
```diff
@@ -438,6 +452,58 @@ data:
...
+supplementalGroups:
+  type: RunAsAny
+volumes:
+- configMap
+- downwardAPI
+- emptyDir
+- persistentVolumeClaim
+- projected
+- secret
+
```
Comment (heypnus) [inline manifest/openshift4/coreos/ncp-openshift4.yaml:505]: sorry I forgot to tell you that this should be removed. It's stale because it's for ocp3.

Comment (timdengyun) [reply manifest/openshift4/coreos/ncp-openshift4.yaml]: Thanks for reminder. corrected.

File: manifest/kubernetes/rhel/ncp-rhel.yaml (hunk 2 above)
Comment (timdengyun) [reply manifest/kubernetes/rhel/ncp-rhel.yaml:735]: Yes. I've tested it, The {{.NcpTolerations}} is equal to {{ .NcpTolerations }}. we have already used this format in replicas:{{.NcpReplicas}}

File: deploy/openshift4/role.yaml [hunk 4]
```diff
@@ -60,3 +60,6 @@ rules:
 - apiGroups: ['']
   resources: [pods/exec]
   verbs: [get, create]
+- apiGroups: ['']
+  resources: [events]
+  verbs: [create, get, list, patch, update, delete]
```
Comment (timdengyun) [inline deploy/openshift4/role.yaml:65]: This added events resources has some issue for operator patch role, is doing debug.

File: manifest/kubernetes/rhel/ncp-rhel.yaml (hunk 2 above)
Comment (heypnus) [reply manifest/kubernetes/rhel/ncp-rhel.yaml:735]: Thanks for the confirmation.

**Review by heypnus — APPROVED** [review]

File: deploy/openshift4/role.yaml (hunk 4 above)
Comment (timdengyun) [reply deploy/openshift4/role.yaml:65]: after applying new role.yaml, the ncp-patch-role works well, so there is no issue.

**Pull request merged**